### ISSUE TITLE
Testing DBMS with pre-defined schema

### ISSUE THREAD

**Issue by xinyuliu12** [sqlancer/sqlancer#267]
I was wondering if sqlancer now supports generating random queries based on a pre-defined database schema.

**Comment by mrigger** [comment]
SQLancer does not support this at the moment, but it would be a valuable addition. Implementing this cleanly would take some effort, though. If you want to use an existing database only for some experiments, you could use the following strategy (assuming SQLite, but this works very similarly for other DBMSs):

Rather than creating a new database, open an existing database in `createDatabase()`:
https://github.com/sqlancer/sqlancer/blob/0b7c724743a77c8fdee84fa70262fd0a278b681a/src/sqlancer/sqlite3/SQLite3Provider.java#L285

In `generateDatabase()`, delete/comment out the statements. For SQLite, actually an option `--generate-new-database` exists that does this:

https://github.com/sqlancer/sqlancer/blob/0b7c724743a77c8fdee84fa70262fd0a278b681a/src/sqlancer/sqlite3/SQLite3Provider.java#L183

If you only want to use a pre-existing schema (but generate the content in SQLancer), you can manually create a database with the desired tables and views, change `createDatabase` as explained above, and comment out the `CREATE TABLE` and `CREATE VIEW` generators in `generateDatabase()`, but keep the generation of the other statements.

Does this make sense/work for now?

**Comment by xinyuliu12** [comment]
Yes, it makes sense now! Thank you so much for your detailed explanation!

**Comment by mrigger** [comment]
Great, I'll keep this open, because it might be useful to add this feature in the future!

**Comment by vivekd01** [comment]
Please give me some project to solve the problem.
